pkg/patterns/declarative/pkg/applier/direct.go

```diff
@@ -121,3 +102,52 @@ func (s *staticRESTClientGetter) ToRESTMapper() (meta.RESTMapper, error) {
 	}
 	return s.RESTMapper, nil
 }
+func (s *staticRESTClientGetter) ToRawKubeConfigLoader() clientcmd.ClientConfig {
+	return namespaceStub(s.namespace)
+}
+
+type namespaceStub string
+
+func (n namespaceStub) Namespace() (string, bool, error) {
+	return string(n), false, nil
+}
+
+// below methods should never be called
+func (namespaceStub) RawConfig() (api.Config, error) {
+	panic("implement me")
+}
+
+func (namespaceStub) ClientConfig() (*rest.Config, error) {
+	panic("implement me")
+}
+
+func (namespaceStub) ConfigAccess() clientcmd.ConfigAccess {
+	panic("implement me")
+}
+
+func NewCmdApply(ioStreams genericclioptions.IOStreams) (*cobra.Command, *apply.ApplyOptions) {
```

Comment:
Did you copy some codes from [k/kubectl](https://github.com/kubernetes/kubectl/blob/v0.22.2/pkg/cmd/apply/apply.go#L163)?
IMHO, it's good not to copy, but use modules.
I think `DirectApplier` is implementend with something tricky way.

Recently, we started these topics related with `prune` functionality at #201.
We may have to have discussion at #201.

Reply:
yes, It's copied from `k/kubectl`.  In the `NewCmdApply` function, `ApplyOptions` is never returned, so I didn't find a way to reuse that piece of code. 

In the previous implementation, some codes are copied from [Complete](https://github.com/kubernetes/kubectl/blob/920da0b7c6952cfe2cdbed4c65e9addc0f7be219/pkg/cmd/apply/apply.go#L209) too, with some customize modification, Which can be very easy to make mistakes. I agreed`DirectApplier` is implemented very trickly indeed.